test/JSON/toJSON.js

```diff
@@ -15,12 +15,141 @@ fnc.prototype.toJSON = function() {
+  obj.d = 4;
+  TEST('{"a":1,"b":2,"d":4}', JSON.stringify(obj));
+}
+
+// test getter - enum - delete 'em all
+{
+  var obj = {};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true, configurable: true });
+
+  obj.a = 1;
+  obj.b = 2;
+  delete obj['c']
+  TEST('{"a":1,"b":2}', JSON.stringify(obj));;
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true });
+  obj.d = 4;
+  TEST('{"c":3,"a":1,"b":2,"d":4}', JSON.stringify(obj));
+  obj[9]=19;
+  TEST('{"9":19,"c":3,"a":1,"b":2,"d":4}', JSON.stringify(obj));
+  delete obj[9]
+  TEST('{"c":3,"a":1,"b":2,"d":4}', JSON.stringify(obj));
+  delete obj['a']
+  TEST('{"c":3,"b":2,"d":4}', JSON.stringify(obj));
+}
+
+// test getter
+{
+  var obj = {a:1, b:2};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true });
+
+  obj.d = 4;
+  TEST('{"a":1,"b":2,"c":3,"d":4}', JSON.stringify(obj));;
+}
+
+// test getter - setter
+{
+  var obj = {a:1, b:2};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true }); // should print
+  Object.defineProperty(obj, "d", { set: () => 4, enumerable: true }); // skips undefined
+
+  obj.e = 5;
+  TEST('{"a":1,"b":2,"c":3,"e":5}', JSON.stringify(obj));
+}
+
+// test getter with revive
+{
+  function rev(name, value) {
+    if (value == 3) {
+      return undefined;
+    }
+    else if (value == true) {
+      return 99;
+    }
+    return value;
+  }
+  var obj = {a:1, b:2};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true });
+
+  obj.d = 4;
+  TEST('{"a":99,"b":2,"d":4}', JSON.stringify( JSON.parse(JSON.stringify(obj), rev) ));
+}
+
+// test getter with revive && objectArray
+{
+  function rev(name, value) {
+    if (value == 3) {
+      return undefined;
+    }
+    else if (value == true) {
+      return 99;
+    }
+    return value;
+  }
+  var obj = {a:1, b:2, q:3};
+  obj[9] = 9;
+  obj[44] = 44;
+  Object.defineProperty(obj, "c", { get: () => 7, enumerable: true });
+
+  obj.d = 4;
+
+  obj = JSON.parse(JSON.stringify(obj), rev);
+  TEST('{"9":9,"44":44,"a":99,"b":2,"c":7,"d":4}', JSON.stringify( obj ));
+}
+
+// test getter with revive && objectArray && no-enum
+{
+  function rev(name, value) {
+    if (value == 9) {
+      return undefined;
+    }
+    else if (value == true) {
+      return 99;
+    }
+    return value;
+  }
+  var obj = {a:1, b:2};
+  obj[7] = 8;
+  obj[9] = 9;
+  obj[44] = 44;
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: false });
+
+  obj.d = 4;
+
+  obj = JSON.parse(JSON.stringify(obj), rev);
+  TEST('{"7":8,"44":44,"a":99,"b":2,"d":4}', JSON.stringify( obj ));
+}
+
 // test - hasObjectArray but no elements in it
-var obj = {"a":1};
-obj[0] = 2;
-TEST(JSON.stringify(obj), '{"0":2,"a":1}');
-delete obj[0];
-TEST(JSON.stringify(obj), '{"a":1}');
+{
+  var obj = {"a":1};
+  obj[0] = 2;
+  TEST(JSON.stringify(obj), '{"0":2,"a":1}');
+  delete obj[0];
+  TEST(JSON.stringify(obj), '{"a":1}');
+}
```

Comment:
You may want to merge these tests with the one I added in `stringifyobject.js`, just for simplicity's sake. Also, using the unit test framework may help pinpoint exact problems, since we can actually name the tests there?

Reply:
Well, we had this discussion with Tom on the original PR that I brought this test file.